azdev/operations/linter/rules/parameter_rules.py

```diff
@@ -139,3 +139,26 @@ def _help_contains_queries(help_str, queries):
         if param_help and _help_contains_queries(param_help, queries):
             raise RuleError("An option {} ends with '-id'. Arguments ending with '-id' "
                             "must be guids/uuids and not resource ids.".format(options_list))
+
+
+@ParameterRule(LinterSeverity.HIGH)
+def option_length_too_long(linter, command_name, parameter_name):
+    min_length = None
+    length_threshold = 22  # only 5% argument's length exceed this value.
+    options_list = linter.get_parameter_options(command_name, parameter_name) or []
+    for option in options_list:
+        if isinstance(option, Deprecated) or option.startswith('--__'):
+            return
+        min_length = min(min_length, len(option)) if min_length else len(option)
+    if min_length and min_length > length_threshold:
+        raise RuleError("The lengths of all options {} are longer than {} ".format(options_list, length_threshold))
```

Comment:
So this means at least one option not exceeding the length_threshold should be provided? Can you add the rule in the error message to make it clear?

Reply:
Sure